Lib/ntpath.py

```diff
@@ -212,15 +239,13 @@ def split(p):
     Either part may be empty."""
     p = os.fspath(p)
     seps = _get_bothseps(p)
-    d, p = splitdrive(p)
+    d, r, p = splitroot(p)
     # set i to index beyond p's last slash
     i = len(p)
     while i and p[i-1] not in seps:
         i -= 1
     head, tail = p[:i], p[i:]  # now tail has no slashes
-    # remove trailing slashes from head, unless it's all slashes
-    head = head.rstrip(seps) or head
-    return d + head, tail
+    return d + r + head.rstrip(seps), tail
```

Comment:
This makes me wonder if `splitroot` shouldn't ensure that `tail` does not start with slashes anymore...?

I guess I'm not getting something, because this behaviour is tested explicitly, but I'm wondering what that's for...

Reply:
Given `p` is path-like, it's guaranteed that `type(fspath(p))().join(splitroot(p)) == fspath(p)`.

The `head` and `tail` values here come from splitting the value of `p` that's assigned by `d, r, p = splitroot(p)`. Say the input path is "D:/eggs//spam". Then `head` is set to "eggs//", and `tail` is "spam". The return value is `('D:/eggs', 'spam')`.

Reply:
If I were writing this function fresh, I'd probably name the `tail` variable `name` or `base` or something like that, so that it doesn't conflict with the documented return value of `splitdrive()` / `splitroot()`. But I'd rather not change the naming of these variables now unless a core dev asks me to, as it adds headaches for `git blame` users.

Reply:
> Drive-by comments from seeing this PR on discourse, feel free to disregard if I'm saying/asking something stupid.

Thanks for the review! All good feedback I think!